functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/ProvidedWorkspace.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2016-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package com.redhat.che.selenium.core.workspace;
+
+import static java.lang.String.format;
+
+import com.redhat.che.selenium.core.client.RhCheTestWorkspaceServiceClient;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicReference;
+import org.eclipse.che.api.core.model.workspace.Workspace;
+import org.eclipse.che.selenium.core.user.TestUser;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** @author Katerina Foniok */
+public class ProvidedWorkspace implements TestWorkspace {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ProvidedWorkspace.class);
+
+  private CompletableFuture<Void> future;
+  private AtomicReference<String> id;
+  private AtomicReference<String> workspaceName;
```

Comment:
Is there a reason why we have this as AtomicReferences, instead of pure String?

Reply:
Not sure - taken from upstream project setting. If you want me to remove it, it needs to be done in RhCheTestWorkspaceImpl too.

Reply:
@Katka92 Upstream equivalent of this class is `org.eclipse.che.selenium.core.workspace.TestWorkspaceImpl` which has only `id` as AtomicReference.  Because `id` is the only field, which is written into. Other fields are only read (after the initial injection) and that may be the reason...

Reply:
:+1: Refactored.